src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -0,0 +1,198 @@
+void OnAddWiFiNetworkCommandCallback(app::Command *, EndpointId, const uint8_t * ssid, const uint8_t * credentials,
+                                     uint64_t breadcrumb, uint32_t timeoutMs)
+{
+    CHIP_ERROR err = CHIP_ERROR_NO_MEMORY;
+
+    for (size_t i = 0; i < kMaxNetworks; i++)
+    {
+        if (sNetworks[i].mNetworkType == NetworkType::kUndefined)
+        {
+            // Assume they are null terminated, see notes above.
+            size_t ssid_Len        = strlen(reinterpret_cast<const char *>(ssid));
+            size_t credentials_Len = strlen(reinterpret_cast<const char *>(credentials));
+            memcpy(sNetworks[i].mData.mWiFi.mSSID, ssid, ssid_Len);
+            sNetworks[i].mData.mWiFi.mSSIDLen = ssid_Len;
+            memcpy(sNetworks[i].mData.mWiFi.mCredentials, credentials, credentials_Len);
+            sNetworks[i].mData.mWiFi.mCredentialsLen = credentials_Len;
+            memcpy(sNetworks[i].mNetworkID, sNetworks[i].mData.mWiFi.mSSID, ssid_Len);
+            sNetworks[i].mNetworkIDLen = ssid_Len;
+
+            sNetworks[i].mNetworkType = NetworkType::kWiFi;
+            sNetworks[i].mEnabled     = false;
+
+            err = CHIP_NO_ERROR;
+            break;
+        }
+    }
+
+    SuccessOrExit(err);
+
+    ChipLogDetail(Zcl, "WiFi provisioning data: SSID: %s", ssid);
+exit:
+    // TODO: We should encode response command here.
+    ChipLogFunctError(err);
+}
+
+namespace {
+CHIP_ERROR DoEnableNetwork(NetworkInfo & network)
+{
+    switch (network.mNetworkType)
+    {
+    case NetworkType::kThread:
+        // TODO(#5040): Add support for thread network provisioning.
+        return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+    case NetworkType::kWiFi:
+#if defined(CHIP_DEVICE_LAYER_TARGET)
+    {
+        DeviceLayer::DeviceNetworkProvisioningDelegateImpl deviceDelegate;
+        return deviceDelegate.ProvisionWiFi(reinterpret_cast<const char *>(network.mData.mWiFi.mSSID),
+                                            reinterpret_cast<const char *>(network.mData.mWiFi.mCredentials));
+    }
+#else
+        return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE);
+#endif
+    break;
+    case NetworkType::kEthernet:
+    case NetworkType::kUndefined:
+    default:
+        return CHIP_ERROR_NOT_IMPLEMENTED;
+    }
+}
+} // namespace
+
+void OnEnableNetworkCommandCallback(app::Command *, EndpointId, const uint8_t * networkID, uint64_t breadcrumb, uint32_t timeoutMs)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    size_t networkSeq;
+    size_t networkID_Len = strlen(reinterpret_cast<const char *>(networkID));
+
+    err = CHIP_ERROR_KEY_NOT_FOUND;
+
+    for (networkSeq = 0; networkSeq < kMaxNetworks; networkSeq++)
+    {
+        if (sNetworks[networkSeq].mNetworkIDLen == networkID_Len && sNetworks[networkSeq].mNetworkType != NetworkType::kUndefined &&
+            memcmp(sNetworks[networkSeq].mNetworkID, networkID, networkID_Len) == 0)
+        {
+            err = DoEnableNetwork(sNetworks[networkSeq]);
+        }
```

Comment:
Maybe use a `NetworkInfo *` to temporary hold `sNetworks[networkSeq]` that would be easier to read

Reply:
You also needs to update the `mEnabled` state of the `NetworkInfo` struct per spec.

Reply:
Alghouth it should be updated via attribute storage, I just update it in DoEnableNetwork as a reminder.